File: benchmarks/lcs.py
m, n = 500, 480
a = [i % 26 for i in range(m)]
b = [(i * 5 + 3) % 26 for i in range(n)]
w = n + 1
dp = [0] * ((m + 1) * w)
for i in range(1, m + 1):
    for j in range(1, n + 1):
        if a[i-1] == b[j-1]:
            dp[i*w+j] = dp[(i-1)*w+(j-1)] + 1
        else:
            dp[i*w+j] = max(dp[(i-1)*w+j], dp[i*w+(j-1)])
print(dp[m*w+n])

File: benchmarks/cpp/lcs.cpp
#include <cstdio>
#include <vector>
#include <algorithm>
int main() {
    const int m = 500, n = 480, w = n + 1;
    std::vector<int> a(m), b(n), dp((m+1)*w, 0);
    for (int i = 0; i < m; i++) a[i] = i % 26;
    for (int i = 0; i < n; i++) b[i] = (i*5+3) % 26;
    for (int i = 1; i <= m; i++)
        for (int j = 1; j <= n; j++)
            dp[i*w+j] = (a[i-1]==b[j-1]) ? dp[(i-1)*w+(j-1)]+1
                                           : std::max(dp[(i-1)*w+j], dp[i*w+(j-1)]);
    printf("%d\n", dp[m*w+n]);
}
